### PR TITLE
[BugFix] Fix column name resolved ignore resolve db name

### LINKED ISSUE #12959: incorrect query plan generated when mysql external table and olap table join
### Steps to reproduce the behavior (Required)
1.  create olap table
```sql
create database ssb_100g;
CREATE TABLE `lineorder_flat` (
  `lo_orderdate` date NOT NULL COMMENT "",
  `lo_orderkey` bigint(20) NOT NULL COMMENT "",
  `lo_linenumber` bigint(20) NOT NULL COMMENT "",
  `lo_custkey` bigint(20) NOT NULL COMMENT "",
  `lo_partkey` bigint(20) NOT NULL COMMENT "",
  `lo_suppkey` bigint(20) NOT NULL COMMENT "",
  `lo_orderpriority` varchar(100) NOT NULL COMMENT "",
  `lo_shippriority` bigint(20) NOT NULL COMMENT "",
  `lo_quantity` bigint(20) NOT NULL COMMENT "",
  `lo_extendedprice` bigint(20) NOT NULL COMMENT "",
  `lo_ordtotalprice` bigint(20) NOT NULL COMMENT "",
  `lo_discount` bigint(20) NOT NULL COMMENT "",
  `lo_revenue` bigint(20) NOT NULL COMMENT "",
  `lo_supplycost` bigint(20) NOT NULL COMMENT "",
  `lo_tax` bigint(20) NOT NULL COMMENT "",
  `lo_commitdate` date NOT NULL COMMENT "",
  `lo_shipmode` varchar(100) NOT NULL COMMENT "",
  `c_name` varchar(100) NOT NULL COMMENT "",
  `c_address` varchar(100) NOT NULL COMMENT "",
  `c_city` varchar(100) NOT NULL COMMENT "",
  `c_nation` varchar(100) NOT NULL COMMENT "",
  `c_region` varchar(100) NOT NULL COMMENT "",
  `c_phone` varchar(100) NOT NULL COMMENT "",
  `c_mktsegment` varchar(100) NOT NULL COMMENT "",
  `s_name` varchar(100) NOT NULL COMMENT "",
  `s_address` varchar(100) NOT NULL COMMENT "",
  `s_city` varchar(100) NOT NULL COMMENT "",
  `s_nation` varchar(100) NOT NULL COMMENT "",
  `s_region` varchar(100) NOT NULL COMMENT "",
  `s_phone` varchar(100) NOT NULL COMMENT "",
  `p_name` varchar(100) NOT NULL COMMENT "",
  `p_mfgr` varchar(100) NOT NULL COMMENT "",
  `p_category` varchar(100) NOT NULL COMMENT "",
  `p_brand` varchar(100) NOT NULL COMMENT "",
  `p_color` varchar(100) NOT NULL COMMENT "",
  `p_type` varchar(100) NOT NULL COMMENT "",
  `p_size` bigint(20) NOT NULL COMMENT "",
  `p_container` varchar(100) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`lo_orderdate`, `lo_orderkey`)
COMMENT "OLAP"
PARTITION BY RANGE(`lo_orderdate`)
(PARTITION p1992 VALUES [('1992-01-01'), ('1993-01-01')),
PARTITION p1993 VALUES [('1993-01-01'), ('1994-01-01')),
PARTITION p1994 VALUES [('1994-01-01'), ('1995-01-01')),
PARTITION p1995 VALUES [('1995-01-01'), ('1996-01-01')),
PARTITION p1996 VALUES [('1996-01-01'), ('1997-01-01')),
PARTITION p1997 VALUES [('1997-01-01'), ('1998-01-01')),
PARTITION p1998 VALUES [('1998-01-01'), ('1999-01-01')))
DISTRIBUTED BY HASH(`lo_orderkey`) BUCKETS 48
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false",
"compression" = "LZ4"
);
```

2. create external table
```sql
create database test;
CREATE EXTERNAL TABLE test.lineorder_flat
(
  `lo_orderdate` date NOT NULL COMMENT "",
  `lo_orderkey` bigint(20) NOT NULL COMMENT "",
  `lo_linenumber` bigint(20) NOT NULL COMMENT "",
  `lo_custkey` bigint(20) NOT NULL COMMENT "",
  `lo_partkey` bigint(20) NOT NULL COMMENT "",
  `lo_suppkey` bigint(20) NOT NULL COMMENT "",
  `lo_orderpriority` varchar(100) NOT NULL COMMENT "",
  `lo_shippriority` bigint(20) NOT NULL COMMENT "",
  `lo_quantity` bigint(20) NOT NULL COMMENT "",
  `lo_extendedprice` bigint(20) NOT NULL COMMENT "",
  `lo_ordtotalprice` bigint(20) NOT NULL COMMENT "",
  `lo_discount` bigint(20) NOT NULL COMMENT "",
  `lo_revenue` bigint(20) NOT NULL COMMENT "",
  `lo_supplycost` bigint(20) NOT NULL COMMENT "",
  `lo_tax` bigint(20) NOT NULL COMMENT "",
  `lo_commitdate` date NOT NULL COMMENT "",
  `lo_shipmode` varchar(100) NOT NULL COMMENT "",
  `c_name` varchar(100) NOT NULL COMMENT "",
  `c_address` varchar(100) NOT NULL COMMENT "",
  `c_city` varchar(100) NOT NULL COMMENT "",
  `c_nation` varchar(100) NOT NULL COMMENT "",
  `c_region` varchar(100) NOT NULL COMMENT "",
  `c_phone` varchar(100) NOT NULL COMMENT "",
  `c_mktsegment` varchar(100) NOT NULL COMMENT "",
  `s_name` varchar(100) NOT NULL COMMENT "",
  `s_address` varchar(100) NOT NULL COMMENT "",
  `s_city` varchar(100) NOT NULL COMMENT "",
  `s_nation` varchar(100) NOT NULL COMMENT "",
  `s_region` varchar(100) NOT NULL COMMENT "",
  `s_phone` varchar(100) NOT NULL COMMENT "",
  `p_name` varchar(100) NOT NULL COMMENT "",
  `p_mfgr` varchar(100) NOT NULL COMMENT "",
  `p_category` varchar(100) NOT NULL COMMENT "",
  `p_brand` varchar(100) NOT NULL COMMENT "",
  `p_color` varchar(100) NOT NULL COMMENT "",
  `p_type` varchar(100) NOT NULL COMMENT "",
  `p_size` bigint(20) NOT NULL COMMENT "",
  `p_container` varchar(100) NOT NULL COMMENT ""
)
ENGINE=mysql
PROPERTIES
(
    "host" = "127.0.0.1",
    "port" = "29030",
    "user" = "root",
    "password" = "",
    "database" = "ssb_100g",
    "table" = "lineorder_flat"
);
```

p.s. The content is too long, please continue to read the comment
### StarRocks version (Required)
main 92db5880c553ddfa464795b877ed94a2851c5afc

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12959 

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
## Please check the version labels which this pr will be auto backported to target branch and click the checkbox:

- [ ] check done.

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 45 / 51 (88.24%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlWithIdUtils.java|0|3|00.00%|[172, 193, 204]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|4|6|66.67%|[170, 251]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|10|11|90.91%|[735]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Field.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|8|8|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|7|7|100.00%|[]|

**Comment by HangyuanLiu** [comment]
run starrocks_be_unittest

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Field.java [hunk 1]
```diff
@@ -169,11 +168,16 @@ public List<Integer> getTmpUsedStructFieldPos() {
         return tmpUsedStructFieldPos;
     }
 
-    public boolean matchesPrefix(TableName prefix) {
-        if (relationAlias != null) {
-            return relationAlias.getTbl().equals(prefix.getTbl());
+    public boolean matchesPrefix(TableName tableName) {
+        if (tableName.getCatalog() != null && !tableName.getCatalog().equals(relationAlias.getCatalog())) {
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Field.java:172]: Can  relationAlias be null?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Field.java:172]: relationAlias can not be null

    /**
     * TableName of field
     * relationAlias is origin table which table name is explicit, such as t0.a
     * Field come from scope is resolved by scope relation alias,
     * such as subquery alias and table relation name
     */

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 37 / 46 (80.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlWithIdUtils.java|0|3|00.00%|[172, 193, 204]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|0|4|00.00%|[482, 483, 484, 488]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|4|6|66.67%|[170, 251]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Field.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 52 / 58 (89.66%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SqlWithIdUtils.java|0|3|00.00%|[172, 193, 204]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AST2SQL.java|4|6|66.67%|[170, 251]|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|11|12|91.67%|[750]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewAnalyzer.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/SelectAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ViewDefBuilder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Field.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/analysis/TableName.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/InsertPlanner.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|8|8|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
run starrocks_admit_test

**Review by packy92 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.2

**Comment by Astralidea** [comment]
This PR also resolve #12444

**Pull request merged**